RFM9x.device.nut

```diff
@@ -0,0 +1,404 @@
+        if (len > 0xff) return;
+
+        
+        if (_sendFinished) {
+            _sendFinished = false;
+            setMode(RFM9X_STANDBY);
+            _writeReg(RFM9X_REG_DIO_MAPPING1, 0x40);
+            setFifoTxBase(0x00);
+            _writeReg(RFM9X_REG_FIFO_ADDR_PTR, 0x00);
+
+            // Write the data here
+            _writeToTXBuffer(data, len);
+            
+            _writeReg(RFM9X_REG_PAYLOAD_LENGTH, len);
+            setMode(RFM9X_FSTX);
+            imp.sleep(0.0002);
+            _sendFinished = false;
+            setMode(RFM9X_TX);
+        } else {
+            throw "previous send not finished, could not send data";
+        }
+        
+    }
+    
+    function receiveData() {
+        setMode(RFM9X_STANDBY);
+        _writeReg(RFM9X_REG_FIFO_ADDR_PTR, 0x00);
+        _writeReg(RFM9X_REG_DIO_MAPPING1, 0x00);
+        setMode(RFM9X_FSRX);
+        imp.sleep(0.0002);
+        rf.setMode(RFM9X_RXCONTINUOUS);
+    }
+    
+    function setPayloadLength(len) {
+        _writeReg(RFM9X_REG_PAYLOAD_LENGTH, len);
+    }
+    
+    function setImplicitHeaderMode(state) {
+        local cur = _readReg(RFM9X_REG_MODEM_CONFIG1);
+        _writeReg(RFM9X_REG_MODEM_CONFIG1, (cur & 0xfe) | (state ? 1 : 0));
+    }
+    
+    function setPreambleLength(len) {
+        _writeReg(RFM9X_REG_PREAMBLE_MSB, (len >>8) & 0xff);
+        _writeReg(RFM9X_REG_PREAMBLE_LSB, len & 0xff);
+    }
+    
+    function setFifoTxBase(start) {
+        _writeReg(RFM9X_REG_FIFO_TX_BASE_ADDR, start & 0xff);
+    }
+    
+    function setFifoRxBase(start) {
+        _writeReg(RFM9X_REG_FIFO_RX_BASE_ADDR, start & 0xff);
+    }
+    
+    function setCodingRate(cr) {
+        local cur = _readReg(RFM9X_REG_MODEM_CONFIG1);
+        local clear = cur & 0xf1; // exclude bits 3-1
+        _writeReg(RFM9X_REG_MODEM_CONFIG1, clear | (CRTABLE[cr] << 1));
+    }
+    
+    function setRxPayloadCRC(state) {
+        local current = _readReg(RFM9X_REG_MODEM_CONFIG2);
+        _writeReg(RFM9X_REG_MODEM_CONFIG2, (current & 0xfb) | (state ? 0x04 : 0x00));
+    }
+    
+    function setBandwidth(bw) {
+        local current = _readReg(RFM9X_REG_MODEM_CONFIG1);
+        _writeReg(RFM9X_REG_MODEM_CONFIG1, BWTABLE[bw] << 4 | (current & 0x0f));
+    }
+    
+    function setMode(mode) {
+        local current = _readReg(RFM9X_REG_OP_MODE);
+        _writeReg(RFM9X_REG_OP_MODE, mode | (current & 0xf8));
+    }
+        
+    function clearInterrupts() {
+        _writeReg(RFM9X_REG_IRQ_FLAGS, 0xff);
+    }
+    
+    function maskAllInterrupts() {
+        _writeReg(RFM9X_REG_IRQ_FLAGS_MASK, 0xff);
+    }
+    
+    // Enable a specific interrupt. Avaliable interrupts are in
+    // RFM9X_FLAGS
+    function enableInterrupt(mask) {
+        local currentInterrupts = _readReg(RFM9X_REG_IRQ_FLAGS);
+        // Mask the interrupt by writing high, therefore enable by writing low
+        local newInterrupts = currentInterrupts & (~ (1<<mask));
```

Comment:
This always clears the `mask` bit in the `currentInterrupts` value. Is that intentional?

I don't see a routine that clears it.